tools/swift-inspect/Sources/SwiftInspectLinux/PTrace.swift

```diff
@@ -0,0 +1,151 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import LinuxSystemHeaders
+
+public class PTrace {
+  enum PTraceError: Error {
+    case ptraceFailure(_ command: Int32, pid: pid_t, errno: Int32 = get_errno())
+    case waitFailure(pid: pid_t, errno: Int32 = get_errno())
+    case unexpectedWaitStatus(pid: pid_t, status: Int32, sigInfo: siginfo_t? = nil)
+  }
+
+  let pid: pid_t
+
+  // Initializing a PTrace instance attaches to the target process, waits for
+  // it to stop, and leaves it in a stopped state. The caller may resume the
+  // process by calling cont().
+  public init(process pid: pid_t) throws {
+    guard ptrace_attach(pid) != -1 else {
+      throw PTraceError.ptraceFailure(PTRACE_ATTACH, pid: pid)
+    }
+
+    while true {
+      var status: Int32 = 0
+      let result = waitpid(pid, &status, 0)
+      guard result != -1 else {
+        if get_errno() == EINTR { continue }
+        throw PTraceError.waitFailure(pid: pid)
+      }
+
+      precondition(pid == result, "waitpid returned unexpected value \(result)")
+
+      if wIfStopped(status) { break }
+    }
+
+    self.pid = pid
+  }
+
+  deinit { _ = ptrace_detach(self.pid) }
+
+  public func cont() throws {
+    guard ptrace_cont(self.pid) != -1 else {
+      throw PTraceError.ptraceFailure(PTRACE_CONT, pid: self.pid)
+    }
+  }
+
+  public func interrupt() throws {
+    guard ptrace_interrupt(self.pid) != -1 else {
+      throw PTraceError.ptraceFailure(PTRACE_INTERRUPT, pid: self.pid)
+    }
+  }
+
+  public func getSigInfo() throws -> siginfo_t {
+    var sigInfo = siginfo_t()
+    guard ptrace_getsiginfo(self.pid, &sigInfo) != -1 else {
+      throw PTraceError.ptraceFailure(PTRACE_GETSIGINFO, pid: self.pid)
+    }
+
+    return sigInfo
+  }
+
+  public func pokeData(addr: UInt64, value: UInt64) throws {
```

Comment:
`poke(address:value:)` seems sufficent as a name. `peek` and `poke` are well established terms for the operations.

Reply:
Since there's also `PTRACE_POKETEXT`, my preference is to keep the name to avoid ambiguity.

Reply:
> peek and poke are well established terms for the operations.

At least for those of us who remember the 1980s :-D